### PR TITLE
Use primary email in blob ownership panel

### PR DESCRIPTION
**Pull request by cbart**
Part of https://github.com/sourcegraph/sourcegraph/issues/49051

## Test plan

Modify story accordingly.

### CODE REVIEW

File: client/web/src/repo/blob/own/FileOwnershipEntry.tsx [hunk 1]
```diff
@@ -23,7 +23,19 @@ export const FileOwnershipEntry: React.FunctionComponent<Props> = ({ owner, reas
...
+    const findEmail = (): string | undefined => {
+        if (owner.__typename !== 'Person') {
+            return undefined
+        }
+        const primaryEmail: { __typename?: 'UserEmail' | undefined; email: string } | null | undefined =
+            owner?.user?.primaryEmail
+        if (primaryEmail?.__typename === 'UserEmail') {
+            return primaryEmail.email
+        }
+        return owner.email
```
Comment (eseliger) [inline client/web/src/repo/blob/own/FileOwnershipEntry.tsx:35]: found this a bit hard to parse, wdyt about

```suggestion
        if (owner.user && owner.user.primaryEmail !== null) {
            return owner.user.primaryEmail.email
        }
        return owner.email
```

**Review by eseliger — APPROVED** [review]

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (+0.16 kb) | 0.00% (+0.16 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/cb/own-primary-email-ownership-panel/compare-report.html;tab=live_object) for a full comparison between the commits aa9796d and 5d6f968 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: client/web/src/repo/blob/own/FileOwnershipEntry.tsx (hunk 1 above)
Comment (cbart) [reply client/web/src/repo/blob/own/FileOwnershipEntry.tsx]: Yes! Apologies, this is my TypeScript confusion.

With some elvises it parses.

File: client/web/src/repo/blob/own/FileOwnershipEntry.tsx [hunk 2]
```diff
@@ -23,7 +23,17 @@ export const FileOwnershipEntry: React.FunctionComponent<Props> = ({ owner, reas
...
+        if (owner.__typename !== 'Person') {
+            return undefined
+        }
+        if (owner.user?.primaryEmail !== null) {
+            return owner.user?.primaryEmail.email
+        }
+        return owner.email
+    }
+
+    const email = findEmail()
```
Comment (limitedmage) [inline client/web/src/repo/blob/own/FileOwnershipEntry.tsx:36]: nit: maybe we should memoize this?

```suggestion
    const email = useMemo(() => {
        if (owner.__typename !== 'Person') {
            return undefined
        }
        if (owner.user?.primaryEmail !== null) {
            return owner.user?.primaryEmail.email
        }
        return owner.email
    }, [owner])
```

**Review by limitedmage — APPROVED** [review]

File: client/web/src/repo/blob/own/FileOwnershipEntry.tsx (hunk 2 above)
Comment (eseliger) [reply client/web/src/repo/blob/own/FileOwnershipEntry.tsx:36]: It's not computational heavy, so we trade 2 conditionals vs 1 equality check on an array for the deps change check, which is a deep equals, sort of. Do you think that's worth it?

Comment (cbart) [reply client/web/src/repo/blob/own/FileOwnershipEntry.tsx:36]: I'd think it's more matter of style than performance. I'm happy to go either way. To my lay eye the original way looks a little scrappy, and with the memo looks more react :)

Comment (limitedmage) [reply client/web/src/repo/blob/own/FileOwnershipEntry.tsx:36]: I think either way it's fine, `useMemo` has some extra overhead so not sure if it's a perf improvement.
